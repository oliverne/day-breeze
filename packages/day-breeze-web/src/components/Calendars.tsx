import React from 'react'
import { useCalendar } from '../hooks/useCalendar/index'
import { CalDate } from './CalDate'

export function Calendars() {
  const {
    prevMonth,
    thisMonth,
    nextMonth,
    goPrevMonth,
    goNextMonth,
    monthTitle
  } = useCalendar()

  return (
    <div className="calendar calendar-lg">
      <div className="calendar-nav navbar">
        <button
          onClick={goPrevMonth}
          type="button"
          className="btn btn-action btn-link btn-lg"
        >
          <i className="icon icon-arrow-left" />
        </button>
        <div className="navbar-primary">{`${monthTitle} ${thisMonth.year}`}</div>
        <button
          onClick={goNextMonth}
          type="button"
          className="btn btn-action btn-link btn-lg"
        >
          <i className="icon icon-arrow-right" />
        </button>
      </div>
      <div className="calendar-container">
        <div className="calendar-header">
          <CalDate day="Sun" />
          <CalDate day="Mon" />
          <CalDate day="Tue" />
          <CalDate day="Wed" />
          <CalDate day="Thu" />
          <CalDate day="Fri" />
          <CalDate day="Sat" />
        </div>
        <div className="calendar-body">
          {prevMonth.days.map(day => (
            <CalDate day={day} key={day.toString()} prevMonth />
          ))}
          {thisMonth.days.map(day => (
            <CalDate day={day} key={day.toString()} />
          ))}
          {nextMonth.days.map(day => (
            <CalDate day={day} key={day.toString()} nextMonth />
          ))}
        </div>
      </div>
    </div>
  )
}
