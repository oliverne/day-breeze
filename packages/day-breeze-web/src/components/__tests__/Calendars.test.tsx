import React from 'react'
import { render } from '@testing-library/react'
import { Calendars } from '../Calendars'

describe('<Calendars />', () => {
  const actualDate = global.Date

  beforeAll(() => {
    const constantDate = new Date('2020-02-01T04:41:20')
    /* eslint no-global-assign:off */
    ;(global as any).Date = class extends Date {
      constructor() {
        super()
        return constantDate
      }
    }
  })

  afterAll(() => {
    global.Date = actualDate
  })

  test('스냅샷', () => {
    const { asFragment } = render(<Calendars />)

    expect(asFragment()).toMatchSnapshot()
  })
})
