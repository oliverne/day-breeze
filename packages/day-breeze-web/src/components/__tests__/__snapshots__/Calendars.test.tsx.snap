// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendars /> 스냅샷 1`] = `
<DocumentFragment>
  <div
    class="calendar calendar-lg Calendars"
  >
    <div
      class="calendar-nav navbar"
    >
      <button
        class="btn btn-action btn-link btn-lg"
        type="button"
      >
        <i
          class="icon icon-arrow-left"
        />
      </button>
      <div
        class="navbar-primary"
      >
        February 2020
      </div>
      <button
        class="btn btn-action btn-link btn-lg"
        type="button"
      >
        <i
          class="icon icon-arrow-right"
        />
      </button>
    </div>
    <div
      class="calendar-container"
    >
      <div
        class="calendar-header"
      >
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          Sun
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          Mon
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          Tue
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          Wed
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          Thu
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          Fri
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          Sat
        </div>
      </div>
      <div
        class="calendar-body"
      >
        <div
          class="calendar-date prev-month"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            26
          </button>
        </div>
        <div
          class="calendar-date prev-month"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            27
          </button>
        </div>
        <div
          class="calendar-date prev-month"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            28
          </button>
        </div>
        <div
          class="calendar-date prev-month"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            29
          </button>
        </div>
        <div
          class="calendar-date prev-month"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            30
          </button>
        </div>
        <div
          class="calendar-date prev-month"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            31
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            1
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            2
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            3
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            4
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            5
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            6
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            7
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            8
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            9
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            10
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            11
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            12
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            13
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            14
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            15
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            16
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            17
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            18
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            19
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            20
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            21
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            22
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            23
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            24
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            25
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            26
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            27
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            28
          </button>
        </div>
        <div
          class="calendar-date"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            29
          </button>
        </div>
        <div
          class="calendar-date next-month"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            1
          </button>
        </div>
        <div
          class="calendar-date next-month"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            2
          </button>
        </div>
        <div
          class="calendar-date next-month"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            3
          </button>
        </div>
        <div
          class="calendar-date next-month"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            4
          </button>
        </div>
        <div
          class="calendar-date next-month"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            5
          </button>
        </div>
        <div
          class="calendar-date next-month"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            6
          </button>
        </div>
        <div
          class="calendar-date next-month"
          data-testid="calendarDate"
        >
          <button
            class="date-item"
            data-testid="dateItem"
            type="button"
          >
            7
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
